lookout/style/typos/generation.py

```diff
@@ -198,41 +197,44 @@ def _lookup_corrections_for_token(self, typo_info: TypoInfo) -> List[Features]:
         candidates = []
         candidate_tokens = self._get_candidate_tokens(typo_info)
         typo_vec = self._vec(typo_info.typo)
-        dist_calc = EditDistance(typo_info.typo, "damerau")
-        for candidate in set(candidate_tokens):
-            candidate_vec = self.wv[candidate]
-            dist = dist_calc.damerau_levenshtein_distance(candidate, self.config["radius"])
+        for candidate, dist in set(candidate_tokens):
             if dist < 0:
                 continue
+            candidate_vec = self._vec(candidate)
             candidates.append(self._generate_features(typo_info, dist, typo_vec,
                                                       candidate, candidate_vec))
         return candidates
 
-    def _get_candidate_tokens(self, typo_info: TypoInfo) -> Set[str]:
-        candidate_tokens = []
+    def _get_candidate_tokens(self, typo_info: TypoInfo) -> Set[Tuple[str, int]]:
+        candidate_tokens = set()
         last_dist = -1
         edit_candidates_count = 0
+        dist_calc = EditDistance(typo_info.typo, "damerau")
         if self.config["edit_dist_number"] > 0:
             for suggestion in self.checker.lookup(typo_info.typo, 2, self.config["max_distance"]):
                 if suggestion.distance != last_dist:
                     edit_candidates_count = 0
                     last_dist = suggestion.distance
                 if edit_candidates_count >= self.config["edit_dist_number"]:
                     continue
-                candidate_tokens.append(suggestion.term)
+                candidate_tokens.add((suggestion.term, suggestion.distance))
                 edit_candidates_count += 1
         if self.config["neighbors_number"] > 0:
             typo_neighbors = self._closest(self._vec(typo_info.typo),
                                            self.config["neighbors_number"])
-            candidate_tokens.extend(typo_neighbors)
+            candidate_tokens |= set([(
+                candidate,
+                dist_calc.damerau_levenshtein_distance(candidate, self.config["radius"]))
```

Comment:
I like more how it was before: `_get_candidate_tokens()` returns a list of tokens and no additional information. 

Now you should think carefully and look in code what the second number means. Let's revert this change.

Reply:
The thing is, the distance calculation is a costly operation, so why repeat it twice

Reply:
do we repeat it twice before?

Reply:
Yep, it's calculated automatically (and probably very cheap too) during the `checker.lookup` for the returned tokens. And then it was calculated separately during the features calculations, so not cool.

Reply:
all right.